### PR TITLE
Update to Babel 7 (beta)

### PR DESCRIPTION
**Pull request by sharkykh**
Only thing I'm not sure about is the configuration for `nyc`.
The tests didn't work without requiring `@babel/register`. I added the other ones too, but I don't know if they are needed.

- Split renovate `packageRules` per language
- Update eslint-plugin-vue to beta version
- Upgrade to Babel 7:
  - `@babel/core`: 7.0.0-rc.1
  - `@babel/plugin-proposal-object-rest-spread`: 7.0.0-rc.1
  - `@babel/polyfill`: 7.0.0-rc.1
  - `@babel/preset-env`: 7.0.0-rc.1
  - `@babel/register`: 7.0.0-rc.1
  - `ava`: 1.0.0-beta.7
  - `babel-loader`: 8.0.0-beta.4
  - `babel-plugin-istanbul`: 5.0.1
  - `require-extension-hooks-babel`: 1.0.0-beta.1
  - `nyc`: 13.0.1
- Add babel group to renovate

### CODE REVIEW

File: themes-default/slim/.babelrc [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
   ],
   "plugins": [
-    "transform-object-rest-spread"
-  ]
+    "@babel/plugin-proposal-object-rest-spread"
+  ],
+  "env": {
+    "test": {
+      "plugins": [
+        "istanbul"
```
Comment (OmgImAlexis) [inline themes-default/slim/.babelrc]: What's this needed for?

Comment (sharkykh) [reply themes-default/slim/.babelrc]: 1. I followed https://istanbul.js.org/docs/tutorials/es2015/ and it doesn't collect any coverage without it.
2. You added the dependency in https://github.com/pymedusa/Medusa/pull/4527/commits/0c7e563ab1a59832c91fcdc979e26a82ada83499, but it wasn't being used.
    > A babel plugin that adds istanbul instrumentation to ES6 code.

**Comment by sharkykh** [comment]
The tests were failing and no coverage was collected after the ava/babel update.

**Comment by sharkykh** [comment]
Waiting for #4878 to be merged first.

**Comment by sharkykh** [comment]
@p0psicles 
Not v0.2.8?
It's part of the move to Webpack IMO

**Pull request merged**